### ISSUE TITLE
Is it possible to set up Q&A ourselves?

### ISSUE THREAD

**Issue by chalitguza** [n4ze3m/dialoqbase#138]
Is it possible to set up Q&A ourselves? I think it might be more accurate if we specify questions and answers in the same set. I noticed in the database that the question and answer data are separated into different columns. I’m not sure if this thought process would be better or not. Would it be more efficient if we define question-answer pairs ourselves through the UI? and it is included in the same set of DB

**Comment by ferstpetr777** [comment]
I completely agree with my colleague, this method is already implemented on another project and it would be nice to have such functionality here too. this project is more convenient to use and already has a rag function.     https://docs.dify.ai/advanced/datasets ( In Segmenting in Question & Answer format, the text is summarized into multiple QA pairs )

**Comment by chalitguza** [comment]
> I completely agree with my colleague, this method is already implemented on another project and it would be nice to have such functionality here too. this project is more convenient to use and already has a rag function. https://docs.dify.ai/advanced/datasets ( In Segmenting in Question & Answer format, the text is summarized into multiple QA pairs )

I have tried the "dify" project, and it looks great with a lot of flexibility for users to customize settings as they wish. However, I have noticed that there are still some issues, particularly with the data. Sometimes, the data I've modified, such as specifying keywords from the mentioned features, gets updated when the index is rearranged according to the automatic filtering system, causing the previously edited keywords to disappear entirely. The second issue is that sometimes the data is not usable, possibly due to overly strict filtering. For example, when I upload multiple documents and then test a search for similar keywords, I don't get any results, even though the data is in the database that I've prepared. I think this is crucial.

On the other hand, when I tried "dialoqbase," it was able to find the mentioned keywords, and the bot could provide responses based on the existing database. So, I chose to use "dialoqbase" over "dify." However, if "dialoqbase" has similar features to "dify," such as customizable data preparation and QA, it would be even better. I hope the "dialoqbase" project will continue to improve in the future. I'll be following your progress.
